### PR TITLE
LPS-140170 LPS-140171 [CP 2.0] New key generation - DXP - on prem environments part 2a and 2b

### PR DESCRIPTION
**Pull request by joaosvassis**

### CODE REVIEW

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/GenerateCardLayout/index.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+						<p className="font-weight-normal">
+							{
+								infoSelectedKey.selectedSubscription
+									?.provisionedCount
+							}
+
+							{' of '}
+
+							{infoSelectedKey.selectedSubscription?.quantity}
```
Comment (joanagrego) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/GenerateCardLayout/index.js:58]:
```suggestion
							{
								infoSelectedKey?.selectedSubscription
									?.provisionedCount
							}

							{' of '}

							{infoSelectedKey?.selectedSubscription?.quantity}
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/GenerateCardLayout/index.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+						</p>
+
+						<p className="m-0">Instance Size</p>
+
+						<p className="font-weight-normal m-0">
+							{infoSelectedKey.selectedSubscription?.instanceSize}
+						</p>
```
Comment (joanagrego) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/GenerateCardLayout/index.js:65]:
```suggestion
						<p className="font-weight-normal m-0">
							{infoSelectedKey?.selectedSubscription?.instanceSize}
						</p>
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/pages/RequiredInformation.js [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+									</Button>
+
+									<div className="dropdown-divider"></div>
+								</div>
+							) : (
+								<div className="cp-input-generate-label px-6">
+									<KeySelect
+										minAvaliableKeysCount={
+											infoSelectedKey.selectedSubscription
+												?.provisionedCount
```
Comment (joanagrego) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/pages/RequiredInformation.js:261]:
```suggestion
											infoSelectedKey?.selectedSubscription
												?.provisionedCount
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/pages/SelectSubscription.js [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+						<div className="d-flex justify-content-between mb-2">
+							<div className="mr-3 w-100">
+								<label htmlFor="basicInput">Product</label>
+
+								<div className="cp-select-card position-relative">
+									<ClaySelect
+										className="cp-select-card mr-2"
+										disabled={true}
+									>
+										{subscriptionGroups.map((product) => (
```
Comment (joanagrego) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/pages/SelectSubscription.js:153]:
```suggestion
										{subscriptionGroups?.map((product) => (
```

**Review by joanagrego — APPROVED** [review]
Good job!

**Review by jairmedeiros — APPROVED** [review]

**Comment by jairmedeiros** [comment]
ci:forward

**Comment by jairmedeiros** [comment]
Just starting reviewing :)

**Review by igordmf — COMMENTED** [review]
Good job!

**Pull request closed without merging**